packages/core/src/amazonq/onboardingPage/walkthrough.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import globals from '../../shared/extensionGlobals'
+import { VSCODE_EXTENSION_ID } from '../../shared/extensions'
+import { Commands } from '../../shared/vscode/commands2'
+import vscode from 'vscode'
+
+/**
+ * Show the Amazon Q walkthrough one time forever when the user adds an Amazon Q connection.
+ * All subsequent calls to this do nothing.
+ */
+export async function showAmazonQWalkthroughOnce(
+    state = globals.context.globalState,
+    showWalkthrough = () => openAmazonQWalkthrough.execute()
+) {
+    const hasShownWalkthroughId = 'aws.amazonq.hasShownWalkthrough'
+    const hasShownWalkthrough = state.get(hasShownWalkthroughId, false)
+    if (hasShownWalkthrough) {
+        return
+    }
```

Comment:
Perhaps we need something like `PromptSettings` , but for globalState? Or can we use PromptSettings for this?

Reply:
Yeah, I think we should create something similar for globalState. Theres probably a lot of duplicate stuff we can remove.